src/Server.php

```diff
@@ -217,21 +251,85 @@ private function handleReadRequest(Response $response, string $path, $contents)
             $mimetype = $filesystem->getMimetype($path);
 
             if ($mimetype === self::MIME_TYPE_DIRECTORY) {
-                $listContents = $filesystem->listContents($path, true);
-                $contents = json_encode($listContents);
+                $contents = $this->listDirectoryAsTurtle($path);
                 $response->getBody()->write($contents);
-
+				$response = $response->withHeader("Content-type", "text/turtle");
                 $response = $response->withStatus(200);
             } else {
                 $contents = $filesystem->read($path);
+				$mimetype = $filesystem->getMimetype($path);
 
                 if ($contents !== false) {
                     $response->getBody()->write($contents);
+					$response = $response->withHeader("Content-type", $mimetype);
                     $response = $response->withStatus(200);
                 }
             }
         }
 
         return $response;
     }
+	
+	private function guid() {
+		return sprintf('%04X%04X-%04X-%04X-%04X-%04X%04X%04X', mt_rand(0, 65535), mt_rand(0, 65535), mt_rand(0, 65535), mt_rand(16384, 20479), mt_rand(32768, 49151), mt_rand(0, 65535), mt_rand(0, 65535), mt_rand(0, 65535));
+	}
+
+	private function listDirectoryAsTurtle($path) {
+        $filesystem = $this->filesystem;
+		$listContents = $filesystem->listContents($path);
+		
+		
+		$name = basename($path) . ":";
+
+		$turtle = array(
+			"$name" => array(
+				"a" => array("ldp:BasicContainer", "ldp:Container"),
+				"ldp:contains" => array()
+			)
+		);
+		
+		foreach ($listContents as $item) {
+			switch($item['type']) {
+				case "file":
+					$filename = "<" . $item['basename'] . ">";
+					$turtle[$filename] = array(
+						"a" => array("ldp:Resource")
+					);
+					$turtle[$name]['ldp:contains'][] = $filename;
+				break;
+				case "dir":
+					$filename = "<" . $item['basename'] . ">";
+					$turtle[$filename] = array(
+						"a" => array("ldp:BasicContainer", "ldp:Container")
+					);
+					$turtle[$name]['ldp:contains'][] = $filename;
+				break;
+				default:
+					throw new \Exception("Unknown type", 500);
+				break;
+			}
+		}
+
+		$container = <<< EOF
+@prefix : <#>.
+@prefix $name <>.
+@prefix ldp: <http://www.w3.org/ns/ldp#>.
+
+EOF;
+
+		foreach ($turtle as $name => $item) {
+			$container .= "\n$name\n";
+			$lines = [];
+			foreach ($item as $property => $values) {
+				if (sizeof($values)) {
+					$lines[] = "\t" . $property . " " . implode(", ", $values);
+				}
+			}
+			
+			$container .= implode(";\n", $lines);
+			$container .= ".\n";
```

Comment:
This line can be concatenated to the previous line. In fact, it can even include the first concat a few lines up! 

```
 			$container .= "\n$name\n" . implode(";\n", $lines) . ".\n";
```

Unless you prefer vsprintf():

```
 			$container .= vsprintf("\n%s\n%s.\n", [$name, implode(";\n", $lines)])
 ```

Reply:
I think that having this in separate lines makes it more readable than the concatted or vsprintf versions - I don't mind spending a few more lines of code there to keep it easier to see what is happening.

Reply:
I'm not opposed to multi-line, mind you, I would just suggest not having `$container .= ` on _both_ lines.

But it's not a show-stopper or anything :grin: